rustdesk 1.4.0 build failure with gcc15

```
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc: In function 'long int mkvparser::UnserializeFloat(IMkvReader*, long long int, long long int, double&)':
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:249:7: error: 'uint32_t' does not name a type
warning: webm-sys@1.0.4:   249 |       uint32_t ff;
warning: webm-sys@1.0.4:       |       ^~~~~~~~
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:24:1: note: 'uint32_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
warning: webm-sys@1.0.4:    23 | #include "common/webmids.h"
warning: webm-sys@1.0.4:   +++ |+#include <cstdint>
warning: webm-sys@1.0.4:    24 | 
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:250:45: error: 'uint32_t' was not declared in this scope
warning: webm-sys@1.0.4:   250 |       static_assert(sizeof(float) == sizeof(uint32_t), "");
warning: webm-sys@1.0.4:       |                                             ^~~~~~~~
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:250:45: note: 'uint32_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:253:5: error: 'ff' was not declared in this scope; did you mean 'ffs'?
warning: webm-sys@1.0.4:   253 |     ff = 0;
warning: webm-sys@1.0.4:       |     ^~
warning: webm-sys@1.0.4:       |     ffs
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:268:7: error: 'uint64_t' does not name a type
warning: webm-sys@1.0.4:   268 |       uint64_t dd;
warning: webm-sys@1.0.4:       |       ^~~~~~~~
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:268:7: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:269:46: error: 'uint64_t' was not declared in this scope
warning: webm-sys@1.0.4:   269 |       static_assert(sizeof(double) == sizeof(uint64_t), "");
warning: webm-sys@1.0.4:       |                                              ^~~~~~~~
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:269:46: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
warning: webm-sys@1.0.4: libwebm/mkvparser/mkvparser.cc:272:5: error: 'dd' was not declared in this scope
warning: webm-sys@1.0.4:   272 |     dd = 0;
warning: webm-sys@1.0.4:       |     ^~
warning: webm-sys@1.0.4: ToolExecError: Command LC_ALL="C" "x86_64-pc-linux-gnu-g++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "libwebm" "-O2" "-pipe" "-march=native" "-fdiagnostics-color=always" "-frecord-gcc-switches" "-ggdb" "-fno-rtti" "-std=gnu++11" "-fno-exceptions" "-o" "/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0/target/release/build/webm-sys-e354e680eb8d8104/out/f6c0733b753af51c-mkvparser.o" "-c" "libwebm/mkvparser/mkvparser.cc" with args x86_64-pc-linux-gnu-g++ did not execute successfully (status code exit status: 1).
error: failed to run custom build command for `webm-sys v1.0.4 (/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rust-webm-d2c4d3ac133c7b0e4c0f656da710b48391981e64/src/sys)`

Caused by:
  process didn't exit successfully: `/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0/target/release/build/webm-sys-233b604534cd49a1/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=ffi.cpp
  OUT_DIR = Some(/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0/target/release/build/webm-sys-e354e680eb8d8104/out)
  OPT_LEVEL = Some(3)
  TARGET = Some(x86_64-unknown-linux-gnu)
  HOST = Some(x86_64-unknown-linux-gnu)
  cargo:rerun-if-env-changed=CXX_x86_64-unknown-linux-gnu
  CXX_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CXX_x86_64_unknown_linux_gnu
  CXX_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CXX
  HOST_CXX = None
  cargo:rerun-if-env-changed=CXX
  CXX = Some(x86_64-pc-linux-gnu-g++)
  cargo:rerun-if-env-changed=CC_KNOWN_WRAPPER_CUSTOM
  CC_KNOWN_WRAPPER_CUSTOM = None
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(false)
  CARGO_CFG_TARGET_FEATURE = Some(fxsr,sse,sse2)
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64-unknown-linux-gnu
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_linux_gnu
  CXXFLAGS_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = Some(-O2 -pipe -march=native -fdiagnostics-color=always -frecord-gcc-switches -ggdb)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  CARGO_ENCODED_RUSTFLAGS = Some(-Cstrip=none-Clinker=x86_64-pc-linux-gnu-gcc-Ctarget-feature=-crt-static-Clink-arg=-Wl,-O1-Clink-arg=-Wl,--as-needed-Clink-arg=-Wl,-z,pack-relative-relocs-Clink-arg=-Wl,--defsym=__gentoo_check_ldflags__=0)
  cargo:warning=libwebm/mkvparser/mkvparser.cc: In function 'long int mkvparser::UnserializeFloat(IMkvReader*, long long int, long long int, double&)':
  cargo:warning=libwebm/mkvparser/mkvparser.cc:249:7: error: 'uint32_t' does not name a type
  cargo:warning=  249 |       uint32_t ff;
  cargo:warning=      |       ^~~~~~~~
  cargo:warning=libwebm/mkvparser/mkvparser.cc:24:1: note: 'uint32_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
  cargo:warning=   23 | #include "common/webmids.h"
  cargo:warning=  +++ |+#include <cstdint>
  cargo:warning=   24 | 
  cargo:warning=libwebm/mkvparser/mkvparser.cc:250:45: error: 'uint32_t' was not declared in this scope
  cargo:warning=  250 |       static_assert(sizeof(float) == sizeof(uint32_t), "");
  cargo:warning=      |                                             ^~~~~~~~
  cargo:warning=libwebm/mkvparser/mkvparser.cc:250:45: note: 'uint32_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
  cargo:warning=libwebm/mkvparser/mkvparser.cc:253:5: error: 'ff' was not declared in this scope; did you mean 'ffs'?
  cargo:warning=  253 |     ff = 0;
  cargo:warning=      |     ^~
  cargo:warning=      |     ffs
  cargo:warning=libwebm/mkvparser/mkvparser.cc:268:7: error: 'uint64_t' does not name a type
  cargo:warning=  268 |       uint64_t dd;
  cargo:warning=      |       ^~~~~~~~
  cargo:warning=libwebm/mkvparser/mkvparser.cc:268:7: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
  cargo:warning=libwebm/mkvparser/mkvparser.cc:269:46: error: 'uint64_t' was not declared in this scope
  cargo:warning=  269 |       static_assert(sizeof(double) == sizeof(uint64_t), "");
  cargo:warning=      |                                              ^~~~~~~~
  cargo:warning=libwebm/mkvparser/mkvparser.cc:269:46: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
  cargo:warning=libwebm/mkvparser/mkvparser.cc:272:5: error: 'dd' was not declared in this scope
  cargo:warning=  272 |     dd = 0;
  cargo:warning=      |     ^~
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 1
  cargo:warning=ToolExecError: Command LC_ALL="C" "x86_64-pc-linux-gnu-g++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "libwebm" "-O2" "-pipe" "-march=native" "-fdiagnostics-color=always" "-frecord-gcc-switches" "-ggdb" "-fno-rtti" "-std=gnu++11" "-fno-exceptions" "-o" "/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0/target/release/build/webm-sys-e354e680eb8d8104/out/f6c0733b753af51c-mkvparser.o" "-c" "libwebm/mkvparser/mkvparser.cc" with args x86_64-pc-linux-gnu-g++ did not execute successfully (status code exit status: 1).
  exit status: 0
  exit status: 0

  --- stderr


  error occurred in cc-rs: Command LC_ALL="C" "x86_64-pc-linux-gnu-g++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "libwebm" "-O2" "-pipe" "-march=native" "-fdiagnostics-color=always" "-frecord-gcc-switches" "-ggdb" "-fno-rtti" "-std=gnu++11" "-fno-exceptions" "-o" "/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0/target/release/build/webm-sys-e354e680eb8d8104/out/f6c0733b753af51c-mkvparser.o" "-c" "libwebm/mkvparser/mkvparser.cc" with args x86_64-pc-linux-gnu-g++ did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
warning: creating a shared reference to mutable static is discouraged
  --> libs/hbb_common/src/platform/mod.rs:65:33
   |
65 |         if let Some(callback) = &GLOBAL_CALLBACK {
   |                                 ^^^^^^^^^^^^^^^^ shared reference to mutable static
   |
   = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
   = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives
   = note: `#[warn(static_mut_refs)]` on by default
help: use `&raw const` instead to create a raw pointer
   |
65 |         if let Some(callback) = &raw const GLOBAL_CALLBACK {
   |                                  +++++++++

warning: `sysinfo` (lib) generated 4 warnings (4 duplicates)
warning: `hbb_common` (lib) generated 4 warnings
 * ERROR: net-misc/rustdesk-1.4.0::gentoo-zh failed (compile phase):
 *   cargo build failed
 * 
 * Call stack:
 *     ebuild.sh, line  136:  Called src_compile
 *   environment, line 2109:  Called cargo_src_compile
 *   environment, line 1003:  Called die
 * The specific snippet of code:
 *       cargo_env "${@}" || die "cargo build failed"
 * 
 * If you need support, post the output of `emerge --info '=net-misc/rustdesk-1.4.0::gentoo-zh'`,
 * the complete build log and the output of `emerge -pqv '=net-misc/rustdesk-1.4.0::gentoo-zh'`.
 * The complete build log is located at '/var/log/portage/net-misc:rustdesk-1.4.0:20250607-040635.log'.
 * For convenience, a symlink to the build log is located at '/var/tmp/portage/net-misc/rustdesk-1.4.0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/net-misc/rustdesk-1.4.0/temp/environment'.
 * Working directory: '/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0'
 * S: '/var/tmp/portage/net-misc/rustdesk-1.4.0/work/rustdesk-1.4.0'
```
https://paste.gentoo.zip/ZpmnLGqr

--- a/net-misc/rustdesk/files/rustdesk-1.4.0-fix-gcc15.patch
+++ b/net-misc/rustdesk/files/rustdesk-1.4.0-fix-gcc15.patch
@@ -1,0 +1,30 @@
+From 3b630045052e1e4d563207ab9e3be8d137c26067 Mon Sep 17 00:00:00 2001
+From: Sergei Trofimovich <slyich@gmail.com>
+Date: Fri, 2 Aug 2024 22:57:35 +0100
+Subject: [PATCH] mkvparser/mkvparser.cc: add missing <cstdint> include
+
+Without the change `libwebm` build fails on upcoming `gcc-15` as:
+
+    [  2%] Building CXX object CMakeFiles/mkvparser.dir/mkvparser/mkvparser.cc.o
+   mkvparser/mkvparser.cc: In function 'long int mkvparser::UnserializeFloat(IMkvReader*, long long int, long long int, double&)':
+    mkvparser/mkvparser.cc:249:7: error: 'uint32_t' does not name a type
+      249 |       uint32_t ff;
+          |       ^~~~~~~~
+
+Change-Id: I6c9ea3ea761f51812f6507f6c928a0915603eb59
+---
+ mkvparser/mkvparser.cc | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/libwebm-libwebm-1.0.0.31/mkvparser/mkvparser.cc b/libwebm-libwebm-1.0.0.31/mkvparser/mkvparser.cc
+index eddbc7eb5..042a0c56c 100644
+--- a/libwebm-libwebm-1.0.0.31/mkvparser/mkvparser.cc
++++ b/libwebm-libwebm-1.0.0.31/mkvparser/mkvparser.cc
+@@ -16,6 +16,7 @@
+ #include <cfloat>
+ #include <climits>
+ #include <cmath>
++#include <cstdint>
+ #include <cstring>
+ #include <memory>
+ #include <new>

--- a/net-misc/rustdesk/rustdesk-1.4.0.ebuild
+++ b/net-misc/rustdesk/rustdesk-1.4.0.ebuild
@@ -58,17 +58,17 @@ RUST_MIN_VER="1.75.0"
 RUST_NEEDS_LLVM=1
 inherit cargo desktop llvm-r1 systemd xdg
 
-DESCRIPTION="An open-source remote desktop, and alternative to TeamViewer."
+DESCRIPTION="An open-source remote desktop, and alternative to TeamViewer"
 HOMEPAGE="https://rustdesk.com/"
-_WEBM_PV="1.0.0.31"
-_VCPKG_COMMIT="2024.11.16"
+_WEBM_TAG="1.0.0.31"
+_VCPKG_TAG="2024.11.16"
 _HWCODEC_EXTERNALS_COMMIT="a0ff168b672ab57c50f09dbe128608e45a1c4a52"
 _HBB_COMMON_COMMIT="6e556f7e1751a3a709cd5cca0df7268ba3cb1c48"
 SRC_URI="
 	https://github.com/rustdesk/rustdesk/archive/refs/tags/${PV}.tar.gz
 		-> ${P}.tar.gz
-	https://distfiles.gentoocn.org/~jinqiang/distfiles/${PN}-1.3.6-vcpkg-${_VCPKG_COMMIT}-lite.tar.gz
-	https://github.com/webmproject/libwebm/archive/refs/tags/libwebm-${_WEBM_PV}.tar.gz
+	https://distfiles.gentoocn.org/~jinqiang/distfiles/${PN}-1.3.6-vcpkg-${_VCPKG_TAG}-lite.tar.gz
+	https://github.com/webmproject/libwebm/archive/refs/tags/libwebm-${_WEBM_TAG}.tar.gz
 	https://github.com/rustdesk/hbb_common/archive/${_HBB_COMMON_COMMIT}.tar.gz
 		-> hbb_common-${_HBB_COMMON_COMMIT}.tar.gz
 	https://github.com/rustdesk-org/externals/archive/${_HWCODEC_EXTERNALS_COMMIT}.tar.gz
@@ -128,21 +128,17 @@ pkg_setup() {
 }
 
 src_prepare() {
-	PATCHES+=(
-		"${FILESDIR}"/rust-sciter.patch
-	)
-	cd "${S}"/.. || die
-
 	default
+	cd "${S}"/.. || die
+	eapply "${FILESDIR}"/rust-sciter.patch
+	eapply "${FILESDIR}/${P}-fix-gcc15.patch"
 
-	cd - || die
+	rm -rf "${S}"/libs/hbb_common || die
+	ln -s "${WORKDIR}"/hbb_common-${_HBB_COMMON_COMMIT} "${S}"/libs/hbb_common || die
 
-	rm -rf libs/hbb_common || die
-	ln -s "${WORKDIR}"/hbb_common-${_HBB_COMMON_COMMIT} libs/hbb_common || die
-
-	cd ../rust-webm-*/src/sys || die
-	rm -rf libwebm/ || die
-	ln -s "${WORKDIR}"/libwebm-libwebm-*/ libwebm || die
+	local _WEBM_COMMIT=`echo "${GIT_CRATES[webm]}" | awk -F';' '{print $2}'`
+	rm -rf "${WORKDIR}/rust-webm-${_WEBM_COMMIT}"/src/sys/libwebm || die
+	ln -s "${WORKDIR}/libwebm-libwebm-${_WEBM_TAG}" "${WORKDIR}/rust-webm-${_WEBM_COMMIT}"/src/sys/libwebm || die
 
 	local _HWCODEC_COMMIT=`echo "${GIT_CRATES[hwcodec]}" | awk -F';' '{print $2}'`
 	rm -rf "${WORKDIR}"/hwcodec-${_HWCODEC_COMMIT}/externals || die

--- a/net-misc/rustdesk/rustdesk-1.4.0.ebuild
+++ b/net-misc/rustdesk/rustdesk-1.4.0.ebuild
@@ -53,7 +53,7 @@ declare -A GIT_CRATES=(
 	[x11]='https://github.com/bjornsnoen/x11-rs;c2e9bfaa7b196938f8700245564d8ac5d447786a;x11-rs-%commit%/x11'
 )
 
-LLVM_COMPAT=( 17 18 19 )
+LLVM_COMPAT=( 17 18 19 20 )
 RUST_MIN_VER="1.75.0"
 RUST_NEEDS_LLVM=1
 inherit cargo desktop llvm-r1 systemd xdg

